starlite/contrib/sqlalchemy/repository.py

```diff
@@ -0,0 +1,540 @@
+            **kwargs: Additional parameters
+
+        Returns:
+            The retrieved instance.
+
+        Raises:
+            RepositoryNotFoundException: If no instance found identified by `item_id`.
+        """
+        with wrap_sqlalchemy_exception():
+            statement = self._filter_select_by_kwargs(statement=self.statement, **{self.id_attribute: item_id})
+            instance = (await self._execute(statement)).scalar_one_or_none()
+            instance = self.check_not_found(instance)
+            self.session.expunge(instance)
+            return instance
+
+    async def get_one(self, **kwargs: Any) -> ModelT:
+        """Get instance identified by ``kwargs``.
+
+        Args:
+            **kwargs: Identifier of the instance to be retrieved.
+
+        Returns:
+            The retrieved instance.
+
+        Raises:
+            RepositoryNotFoundException: If no instance found identified by `item_id`.
+        """
+        with wrap_sqlalchemy_exception():
+            statement = self._filter_select_by_kwargs(statement=self.statement, **kwargs)
+            instance = (await self._execute(statement)).scalar_one_or_none()
+            instance = self.check_not_found(instance)
+            self.session.expunge(instance)
+            return instance
+
+    async def get_one_or_none(self, **kwargs: Any) -> ModelT | None:
+        """Get instance identified by ``kwargs`` or None if not found.
+
+        Args:
+            **kwargs: Identifier of the instance to be retrieved.
+
+        Returns:
+            The retrieved instance or None
+        """
+        with wrap_sqlalchemy_exception():
+            statement = self._filter_select_by_kwargs(statement=self.statement, **kwargs)
+            instance = (await self._execute(statement)).scalar_one_or_none()
+            if instance:
+                self.session.expunge(instance)
+            return instance or None
+
+    async def get_or_create(self, **kwargs: Any) -> tuple[ModelT, bool]:
+        """Get instance identified by ``kwargs`` or create if it doesn't exist.
+
+        Args:
+            **kwargs: Identifier of the instance to be retrieved.
+
+        Returns:
+            a tuple that includes the instance and whether or not it needed to be created.
+        """
+        existing = await self.get_one_or_none(**kwargs)
+        if existing:
+            return (existing, False)
+        return (await self.add(self.model_type(**kwargs)), True)  # type: ignore[arg-type]
+
+    async def count(self, *filters: FilterTypes, **kwargs: Any) -> int:
+        """Get the count of records returned by a query.
+
+        Args:
+            *filters: Types for specific filtering operations.
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            Count of records returned by query, ignoring pagination.
+        """
+        statement = self.statement.with_only_columns(
+            sql_func.count(  # pylint: disable=not-callable
```

Comment:
do the pylint directives work with ruff? or can this be removed?

Reply:
I removed most of the pylint things in the ruff PR, as lots of false positives don’t occur there. If you still need to suppress something, you can just use the `#noqa: ` notation with the pylint error code. The pylint specific notation isn’t needed anymore.

Reply:
fwict, moving to ruff has been a massive positive - and to top it all off, my isort issues are gone :heart:

Reply:
It's a blessing, really